src/apps/loyalty/containers/forgot_password/index.tsx

```diff
@@ -0,0 +1,146 @@
+// import * as fetch from "isomorphic-fetch"
+import * as React from "react"
+import styled from "styled-components"
+
+import Button from "../../../../components/buttons/inverted"
+import Icon from "../../../../components/icon"
+import Input from "../../../../components/input"
+import Message from "../../../../components/message"
+import Text from "../../../../components/text"
+import TextLink from "../../../../components/text_link"
+
+import colors from "../../../../assets/colors"
+
+const Container = styled.div`
+  padding: 20px;
+  max-width: 350px;
+  margin: 0 auto;
+  align-text: center;
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+`
+
+const StyledInput = styled(Input)`
+  margin: 5px 0;
+  width: 100%;
+`
+
+interface Props {
+  submitEmailUrl: string,
+  appToken: string
+}
+
+interface State {
+  email: string
+  error?: boolean
+  showMessage: boolean
+  isEmailValid: boolean
+}
+
+class ForgotPasswordForm extends React.Component<Props, any> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      email: "",
+      showMessage: false,
```

Comment:
curious about this state, I'm not sure why you need it? Is it to handle showing an error or the success message? If so, you could probably just always show that message string if it's not empty or something? #minor, was just confused.